apps/frontend/src/lib/imports/imports.ts

```diff
@@ -186,12 +186,17 @@ export async function refreshData(): Promise<void> {
         }
 
         if (accessToken.expires_at && now > expiresAt) {
-          // TODO: swap to use refresh token
-          const tokenExists = await storage.getOAuthAccessToken(app);
-          if (tokenExists) {
-            storage.deleteOAuthAccessToken(app);
-            // As this toast will only be shown once, keep it
-            toast.error(`${capitalized} token has expired, reauth app to refresh token.`, {duration: 5000});
+          // If token expired, use refresh token, else warn user and delete token
+          try {
+            accessToken = await refreshAndSaveToken(app, details, accessToken);
+          } catch (error) {
+            const tokenExists = await storage.getOAuthAccessToken(app);
+            if (tokenExists) {
+              storage.deleteOAuthAccessToken(app);
+              // As this toast will only be shown once, keep it
+              console.error("Token refresh failed:", errorToString(error));
+              toast.error(`${capitalized} token has expired, reauth app to refresh token.`, {duration: 5000});
```

Comment:
Does this happen in the background or on manual action? If background then we should not display this toast, per previous discussion